channels/applicationinsights-channel-js/src/SendBuffer.ts

```diff
@@ -345,3 +348,23 @@ export class SessionStorageSendBuffer extends BaseSendBuffer implements ISendBuf
         // @DynamicProtoStub -- DO NOT add any code as this will be removed during packaging
     }
 }
+
+
+
+/*
+ * Session storage buffer holds a copy of all unsent items in the browser session storage.
+ */
+export class SessionStorageSendBuffer extends StringStorageSendBuffer implements ISendBuffer {
```

Comment:
If desired, these could be turned into simple factory functions instead. E.g.,
```
export const SessionStorageSendBuffer = (logger: IDiagnosticLogger, config: ISenderConfig) =>
    new StringStorageSendBuffer(logger, config, utlSetSessionStorage, utlGetSessionStorage);
```

This would mean the `setBuffer` argument can be simplified a bit:
```
_self.setBuffer = (bufferFactory: (logger: IDiagnosticLogger, config: ISenderConfig) => ISendBuffer) => {
    _self._buffer = bufferFactory(_self.diagLog(), _self._senderConfig);
}
```

Reply:
Have removed this anyways now.